server/models/User.ts

```diff
@@ -405,6 +405,23 @@ class User extends ParanoidModel {
     );
   };
 
+  /**
+   * Returns a list of teams that have a user matching this user's email.
+   *
+   * @returns A promise resolving to a list of teams
+   */
+  availableTeams = async () => {
+    const res = await (this.constructor as typeof User)
+      .scope("withTeam")
+      .findAll({
+        where: {
+          email: this.email,
+        },
+      });
+
+    return res.map((user) => user.team);
```

Comment:
stylistically, this isn't quite idiomatic (Sequelize or SQL equivalent). 

in SQL
```sql
SELECT * from team 
INNER JOIN users ON user.teamId = team.id 
    AND user.email = "me@foo.com"
```

in Sequelize
```javascript
Team.findAll({
  include: [{
    model: User,
    required: true
    where: {email: "me@foo.com"}
   }]
});

```

Reply:
Why _did_ i do it like this 🤔 There might be a reason that will become clear but I'll try the refactor